### PR TITLE
doc: net: Minor fixes to BSD socket documentation

### PR DESCRIPTION
**Pull request by jukkar**
Some clarifications to BSD socket descriptions. Also added
rendering hints for some strings.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

### CODE REVIEW

**Review by nashif — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]

File: doc/reference/networking/sockets.rst [hunk 1]
```diff
@@ -17,15 +17,13 @@ and port existing simple networking applications to Zephyr.
 Here are the key requirements and concepts which governed BSD Sockets
 compatible API implementation for Zephyr:
 
-* Should have minimal overhead, similar to the requirement for other
+* Has minimal overhead, similar to the requirement for other
```
Comment (pfalcon) [inline doc/reference/networking/sockets.rst]: Did you read intro sentence before this bullet list? It reads:

> Here are the key requirements and concepts which governed BSD Sockets
compatible API implementation for Zephyr:

After this, you cannot say "has", you say "should have".

**Review by pfalcon — CHANGES_REQUESTED** [review]
There're stylistic errors with this change.

File: doc/reference/networking/sockets.rst (hunk 1 above)
Comment (jukkar) [reply doc/reference/networking/sockets.rst]: I disagree. There is no point of saying "should have" here as we have the stuff implemented already.

Comment (pfalcon) [reply doc/reference/networking/sockets.rst]: > I disagree. There is no point of saying "should have" here as we have the stuff implemented already.

Well, it's not important if it's already implemented or not, we talk about *requirements* here. And "should" there was added to draw similarity to usage of modals in RFCs (https://www.ietf.org/rfc/rfc2119.txt).

All in all, I've re-read it again, and stuff like:

> Here are the key requirements and concepts which governed BSD Sockets compatible API implementation for Zephyr:
> * Has ...
> * Is ...

doesn't read like proper English to me. If you really want to do something about "should", then use infinitive phrases:

> Here are the key requirements and concepts which governed BSD Sockets compatible API implementation for Zephyr:
> * To have ...
> * To be ...

Or reword the intro sentence, "Here are the key requirements and concepts which governed BSD Sockets compatible API implementation for Zephyr:".

Also calling @dbkinder for extra consult here.

Comment (jukkar) [reply doc/reference/networking/sockets.rst]: I see no need to say "should" here. The "should" implies that something is optional which is not the case here.

Comment (pfalcon) [reply doc/reference/networking/sockets.rst]: > The "should" implies that something is optional

No, it implies that something is necessary.

But ok, use infinitives then, as required by grammatical/stylistic rules.

**Comment by jukkar** [comment]
Could this be merged, I see no issues in the text?

**Comment by jukkar** [comment]
Could we merge this?

**Pull request merged**